src/main/java/core/basesyntax/BalanceOperation.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
```

Comment:
When implementing the BalanceOperation, it's necessary to ensure that the balance is positive after the operation. Please add a check to ensure that the resulting quantity for the fruit is not negative.